x-pack/plugins/cases/public/components/user_action_tree/helpers.tsx

```diff
@@ -0,0 +1,373 @@
+  getCaseDetailHrefWithCommentId,
+  getRuleDetailsHref,
+  onRuleDetailsClick,
+  index,
+  loadingAlertData,
+  ruleId,
+  ruleName,
+  onShowAlertDetails,
+}: {
+  action: CaseUserActions;
+  getCaseDetailHrefWithCommentId: (commentId: string) => string;
+  getRuleDetailsHref: (ruleId: string | null | undefined) => string;
+  onRuleDetailsClick: (ruleId: string | null | undefined) => void;
+  onShowAlertDetails: (alertId: string, index: string) => void;
+  alertId: string;
+  index: string;
+  loadingAlertData: boolean;
+  ruleId?: string | null;
+  ruleName?: string | null;
+}): EuiCommentProps => {
+  return {
+    username: (
+      <UserActionUsernameWithAvatar
+        username={action.actionBy.username}
+        fullName={action.actionBy.fullName}
+      />
+    ),
+    className: 'comment-alert',
+    type: 'update',
+    event: (
+      <AlertCommentEvent
+        alertId={alertId}
+        getRuleDetailsHref={getRuleDetailsHref}
+        loadingAlertData={loadingAlertData}
+        onRuleDetailsClick={onRuleDetailsClick}
+        ruleId={ruleId}
+        ruleName={ruleName}
+        commentType={CommentType.alert}
+      />
+    ),
+    'data-test-subj': `${action.actionField[0]}-${action.action}-action-${action.actionId}`,
+    timestamp: <UserActionTimestamp createdAt={action.actionAt} />,
+    timelineIcon: 'bell',
+    actions: (
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <UserActionCopyLink
+            id={action.actionId}
+            getCaseDetailHrefWithCommentId={getCaseDetailHrefWithCommentId}
+          />
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <UserActionShowAlert
+            id={action.actionId}
+            alertId={alertId}
+            index={index}
+            onShowAlertDetails={onShowAlertDetails}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    ),
+  };
+};
+
+export const toStringArray = (value: unknown): string[] => {
+  if (Array.isArray(value)) {
+    return value.reduce<string[]>((acc, v) => {
+      if (v != null) {
+        switch (typeof v) {
+          case 'number':
+          case 'boolean':
+            return [...acc, v.toString()];
+          case 'object':
+            try {
+              return [...acc, JSON.stringify(v)];
+            } catch {
+              return [...acc, 'Invalid Object'];
+            }
+          case 'string':
+            return [...acc, v];
+          default:
+            return [...acc, `${v}`];
+        }
+      }
+      return acc;
+    }, []);
+  } else if (value == null) {
+    return [];
+  } else if (!Array.isArray(value) && typeof value === 'object') {
+    try {
+      return [JSON.stringify(value)];
+    } catch {
+      return ['Invalid Object'];
+    }
+  } else {
+    return [`${value}`];
+  }
+};
+
+export const getGeneratedAlertsAttachment = ({
+  action,
+  alertIds,
+  getCaseDetailHrefWithCommentId,
+  getRuleDetailsHref,
+  onRuleDetailsClick,
+  renderInvestigateInTimelineActionComponent,
```

Comment:
we can talk about it more when we pull timeline in, but I think we might want to make this a generic function that renders any component so we can re-purpose for observability needs?